src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -982,15 +970,29 @@ Status ServiceBasedTaskInfoAccessor::AttemptTaskReconstruction(
   return Status::OK();
 }
 
-Status ServiceBasedTaskInfoAccessor::AsyncReSubscribe() {
+void ServiceBasedTaskInfoAccessor::AsyncReSubscribe(bool is_pubsub_server_restarted) {
   RAY_LOG(INFO) << "Reestablishing subscription for task info.";
-  for (auto &item : subscribe_task_operations_) {
-    RAY_CHECK_OK(item.second(nullptr));
-  }
-  for (auto &item : subscribe_task_lease_operations_) {
-    RAY_CHECK_OK(item.second(nullptr));
+  if (is_pubsub_server_restarted) {
```

Comment:
Document this:
···
If only the GCS sever has restarted, we only need to fetch data from the GCS server. If the pub-sub server has also restarted, we need to resubscribe to the pub-sub server first, then fetch data from the GCS server.
```

Reply:
fixed